website/cue/reference/components/sinks.cue

```diff
@@ -287,7 +287,7 @@ components: sinks: [Name=string]: {
 										enum: {
 											bytes:               "Byte frames are concatenated."
 											character_delimited: "Byte frames are delimited by a chosen character."
-											length_delimited:    "Byte frames are encoded by adding a length header."
+											length_delimited:    "Byte frames whose messages are prefixed by a unsized big-endian 32-bit integer indicating the length."
```

Comment:
How does a user supply this integer?

Reply:
Yeah, it'd be up to the client to include it when framing. According to https://docs.rs/tokio-util/0.7.3/tokio_util/codec/length_delimited/index.html this sort of framing is fairly common.

Reply:
I think the user is not supposed to provide the integer because it is calculated by the framer itself.
As the example shows, you provide the data bytes to transmit:
```rust
    let frame = Bytes::from("hello world");
    transport.send(frame).await?;
```
And the framer encodes ("frames") the data accordingly:
```
+---- len: u32 ----+---- data ----+
| \x00\x00\x00\x0b |  hello world |
+------------------+--------------+
```

Reply:
Yeah - was a late night total misunderstanding how things work 😅